nebula-cache-manager/core/classes/CacheManager_Tests.cls

```diff
@@ -367,6 +502,64 @@ private class CacheManager_Tests {
     System.Assert.isTrue(CacheManager.getOrganizationCache().getKeys().isEmpty());
   }
 
+  @IsTest
+  static void it_adds_new_key_to_session_cache_with_cachebuiler_when_session_platform_cache_is_available() {
+    String mockKey = 'SomeKey';
+    System.Type cacheBuilderStubType = CacheBuilderStub.class;
+    MockPlatformCachePartitionProxy mockOrganizationPartitionProxy = new MockPlatformCachePartitionProxy(true);
+    System.Assert.isTrue(mockOrganizationPartitionProxy.isAvailable());
+    CacheManager.setMockPartitionProxy(CacheManager.PlatformCacheType.SESSION, mockOrganizationPartitionProxy);
+    System.Assert.areEqual(0, mockOrganizationPartitionProxy.containsMethodCallCount);
+    System.Assert.isFalse(CacheManager.getSessionCache().contains(mockKey));
+    System.Assert.areEqual(1, mockOrganizationPartitionProxy.containsMethodCallCount);
+    System.Assert.areEqual(0, mockOrganizationPartitionProxy.getCacheBuilderMethodCallCount);
+
+    Object generatedValue = CacheManager.getSessionCache().get(mockKey, cacheBuilderStubType);
+
+    System.Assert.areEqual(1, mockOrganizationPartitionProxy.getCacheBuilderMethodCallCount);
+    System.Assert.isTrue(mockOrganizationPartitionProxy.contains(mockKey));
+    System.Assert.isNotNull(generatedValue);
+
+    Object cachedValue = CacheManager.getSessionCache().get(mockKey, cacheBuilderStubType);
+
+    System.Assert.isNotNull(cachedValue);
+    System.Assert.areEqual(generatedValue, cachedValue);
+    System.Assert.areEqual(1, mockOrganizationPartitionProxy.getCacheBuilderMethodCallCount);
+    System.Assert.areEqual(cachedValue, mockOrganizationPartitionProxy.get(mockKey));
+    System.Assert.isTrue(mockOrganizationPartitionProxy.contains(mockKey));
+    System.Assert.isTrue(CacheManager.getSessionCache().contains(mockKey));
+    System.Assert.areEqual(cachedValue, CacheManager.getSessionCache().get(mockKey));
+  }
+
+  @IsTest
+  static void it_adds_new_key_to_fallback_transaction_cache_with_cachebuiler_when_session_platform_cache_is_not_available() {
```

Comment:
Same here, there's a small typo here - the word `cachebuilder` is missing the letter 'd'